### PR TITLE
samples update

### PR DESCRIPTION
**Pull request by rachfop**
Signed-off-by: Patrick Rachford <prachford@icloud.com>

## What does this PR do?

Adds samples.

## Java

how-to-add-custom-search-attributes-to-workflow-executions-at-start-time-in-java.md
https://github.com/temporalio/documentation/issues/1475
how-to-asynchronously-complete-an-activity-in-java.md
https://github.com/temporalio/documentation/issues/1456
how-to-configure-tracing-in-java.md
https://github.com/temporalio/documentation/issues/1469
how-to-log-from-a-workflow-in-java.md
how-to-remove-search-attributes-from-a-workflow-in-java.md
how-to-set-a-custom-logger-in-java.md
https://github.com/temporalio/documentation/issues/1471
how-to-upsert-custom-search-attributes-to-a-workflow-executions-during-execution-in-java.md
https://github.com/temporalio/documentation/issues/1477

## Python
how-to-configure-tracing-in-python.md
https://github.com/temporalio/documentation/issues/1467
how-to-send-a-query-to-a-workflow-execution-in-python.md
how-to-set-a-parent-close-policy-in-python.md

## TypeScript
how-to-asynchronously-complete-an-activity-in-typescript.md
https://github.com/temporalio/documentation/issues/1455
how-to-set-a-parent-close-policy-in-typescript.md
https://github.com/temporalio/documentation/issues/1461

## Missing
How-to-send-a-query-to-a-workflow-in-php
How-to-emit-metrics-in-php
How-to-configure-tracing-in-php
How-to-set-a-custom-logger-in-php

How-to-list-workflow-executions-using-the-client-in-java

### CODE REVIEW

**Review by flossypurse — COMMENTED** [review]
Overall, awesome.
Found 1 content and metadata misalignment that needs fixing though.

File: docs-src/python/how-to-asynchronously-complete-an-activity-in-python.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+id: how-to-asynchronously-complete-an-activity-in-python
+title: How to asynchronously complete an Activity in Python
+sidebar_label: Asynchronous Activity Completion
+description: Enables the Activity Function to return without the Activity Execution completing.
```
Comment (djmagee) [inline docs-src/python/how-to-asynchronously-complete-an-activity-in-python.md:5]:
```suggestion
description: To enable an Activity Function to return without the Activity Execution completing, call the `cancel` task object.
```

File: docs-src/python/how-to-configure-tracing-in-python.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+id: how-to-configure-tracing-in-python
+title: How to configure tracing in Python
+sidebar_label: Configure tracing
+description: Configure tracing
+tags:
+  - python
+  - how-to
+---
+
+To configure tracing in python, install the ``opentelemetry` dependencies.
```
Comment (djmagee) [inline docs-src/python/how-to-configure-tracing-in-python.md:11]:
```suggestion
To configure tracing in Python, install the `opentelemetry` dependencies.
```

File: docs-src/python/how-to-configure-tracing-in-python.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+```bash
+# This command installs the `opentelemtry` dependencies.
```
Comment (djmagee) [inline docs-src/python/how-to-configure-tracing-in-python.md:14]:
```suggestion
# This command installs the `opentelemetry` dependencies.
```

File: docs-src/typescript/how-to-asynchronously-complete-an-activity-in-typescript.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+---
+id: how-to-asynchronously-complete-an-activity-in-typescript
+title: How to asynchronous complete an Activity in TypeScript
+sidebar_label: Asynchronous complete an Activity
+description: Asynchronous complete an Activity in TypeScript with the complete method.
```
Comment (djmagee) [inline docs-src/typescript/how-to-asynchronously-complete-an-activity-in-typescript.md:5]:
```suggestion
title: How to asynchronously complete an Activity in TypeScript
sidebar_label: Asynchronously complete an Activity
description: To asynchronously complete an Activity in TypeScript, set `AsyncCompletionClient` to `complete`.
```

File: docs-src/typescript/how-to-asynchronously-complete-an-activity-in-typescript.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+tags:
+  - typescript
+  - how-to
+---
+
+Set the [`AsyncCompletionClient`](https://typescript.temporal.io/api/classes/client.AsyncCompletionClient) class to the [`complete`](https://typescript.temporal.io/api/classes/client.AsyncCompletionClient#complete) method, to asynchronous complete an Activity with a Task Token.
```
Comment (djmagee) [inline docs-src/typescript/how-to-asynchronously-complete-an-activity-in-typescript.md:11]:
```suggestion
To asynchronously complete an Activity with a Task Token, set the [`AsyncCompletionClient`](https://typescript.temporal.io/api/classes/client.AsyncCompletionClient) class to the [`complete`](https://typescript.temporal.io/api/classes/client.AsyncCompletionClient#complete) method.
```

File: docs-src/typescript/how-to-set-a-parent-close-policy-in-typescript.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+title: How to set Parent Close Policy in Typescript
+sidebar_label: Parent Close Policy
+description: Use the `parentClosePolicy` property.
+tags:
+  - typescript
+  - developer-guide
+  - how-to
+---
+
+Use the [`parentClosePolicy`](https://typescript.temporal.io/api/interfaces/workflow.ChildWorkflowOptions#parentclosepolicy) property to specify how the Child reacts to a Parent Workflow reaching a `Closed` state.
```
Comment (djmagee) [inline docs-src/typescript/how-to-set-a-parent-close-policy-in-typescript.md:12]:
```suggestion
To specify how a Child Workflow reacts to a Parent Workflow reaching a Closed state, use the [`parentClosePolicy`](https://typescript.temporal.io/api/interfaces/workflow.ChildWorkflowOptions#parentclosepolicy) property.
```

File: docs/java/how-to-add-custom-search-attributes-to-workflow-executions-at-start-time-in-java.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+To set a custom Search Attribute, call the [`setSearchAttributes()`](<https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/client/WorkflowOptions.Builder.html#setSearchAttributes(java.util.Map)>) method.
+
+```java
+    WorkflowOptions workflowOptions =
+        WorkflowOptions.newBuilder()
+            .setSearchAttributes(generateSearchAttributes())
+            .build();
+```
+
+Where `generateSearchAttributes()` is a `Map<String, ?>` from the search attribute as a key to a value of one of the following types.
```
Comment (djmagee) [inline docs/java/how-to-add-custom-search-attributes-to-workflow-executions-at-start-time-in-java.md:21]:
```suggestion
`generateSearchAttributes()` is a `Map<String, ?>` from the Search Attribute used as the key to a value of one of the following types.
```

File: docs/java/how-to-add-custom-search-attributes-to-workflow-executions-at-start-time-in-java.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+- `String`
+- `long`
```
Comment (djmagee) [inline docs/java/how-to-add-custom-search-attributes-to-workflow-executions-at-start-time-in-java.md:24]: Is `long` indeed lowercase?

File: docs/java/how-to-asynchronously-complete-an-activity-in-java.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
+---
+id: how-to-asynchronously-complete-an-activity-in-java
+title: How to asynchronous complete an Activity in Java
+sidebar_label: Asynchronous complete an Activity
+description: Asynchronous complete an Activity in Java.
```
Comment (djmagee) [inline docs/java/how-to-asynchronously-complete-an-activity-in-java.md:5]:
```suggestion
title: How to asynchronously complete an Activity in Java
sidebar_label: Asynchronously complete an Activity
description: To asynchronously complete an Activity in Java, set the `ActivityCompletionClient` interface to the `complete()` method.
```

File: docs/java/how-to-asynchronously-complete-an-activity-in-java.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+      ForkJoinPool.commonPool().execute(() -> composeGreetingAsync(taskToken, greeting, name));
+      context.doNotCompleteOnReturn();
+
+      // Since we have set doNotCompleteOnReturn(), the workflow action method return value is
+      // ignored.
+      return "ignored";
+    }
+```
+
+When this method is called during an Activity Execution, the Activity Execution does not complete when it's method returns.
```
Comment (djmagee) [inline docs/java/how-to-asynchronously-complete-an-activity-in-java.md:72]:
```suggestion
When this method is called during an Activity Execution, the Activity Execution does not complete when its method returns.
```

File: docs/java/how-to-configure-tracing-in-java.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
+---
+id: how-to-configure-tracing-in-java
+title: How to configure tracing in Java
+sidebar_label: Configure tracing
+description: Configure tracing
```
Comment (djmagee) [inline docs/java/how-to-configure-tracing-in-java.md:5]:
```suggestion
description: To configure tracing in Java, register the `OpenTracingClientInterceptor()` interceptor.
```

File: docs/java/how-to-configure-tracing-in-java.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+tags:
+  - java
+  - how-to
+---
+
+To configure tracing in Java, register the `OpenTracingClientInterceptor()` interceptor. You can register the interceptors on both the Temporal Client side and on the Worker side.
```
Comment (djmagee) [inline docs/java/how-to-configure-tracing-in-java.md:11]:
```suggestion
To configure tracing in Java, register the `OpenTracingClientInterceptor()` interceptor.
You can register the interceptors on both the Temporal Client side and the Worker side.
```

**Review by djmagee — APPROVED** [review]

**Review by flossypurse — APPROVED** [review]
Just gotta resolve the merge conflicts ;)

File: docs-src/python/how-to-asynchronously-complete-an-activity-in-python.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+id: how-to-asynchronously-complete-an-activity-in-python
+title: How to asynchronously complete an Activity in Python
+sidebar_label: Asynchronous Activity Completion
+description: To enable an Activity Function to return without the Activity Execution completing, call the `cancel` task object.
+tags:
+  - how-to
+  - python
+---
+
+To cancel an asynchronous Activity, call the [`cancel`](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.cancel) task object.
```
Comment (cretz) [inline docs-src/python/how-to-asynchronously-complete-an-activity-in-python.md:11]: Asynchronous activity completion is unrelated to cancellation. I think this code and documentation isn't what should be present for "Asynchronous Activity Completion". Asynchronous activity completion is about returning from an activity without completing it and then completing it later (I put code in #1454 relating to each of those two parts).

File: docs-src/python/how-to-set-a-parent-close-policy-in-python.md [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+title: How to set a Parent Close Policy in Python
+sidebar_label: Parent Close Policy
+description: Create an instance of the `ParentClosePolicy` class.
+tags:
+  - python
+  - developer-guide
+  - how-to
+---
+
+Create an instance of the [`ParentClosePolicy`](https://python.temporal.io/temporalio.workflow.ParentClosePolicy.html) class and specify a constant to determine how a Child Workflow should be handled when the Parent closes.
```
Comment (cretz) [inline docs-src/python/how-to-set-a-parent-close-policy-in-python.md:12]: You don't create an instance `ParentClosePolicy`, you just set the `parent_close_policy` parameter.

Not sure code so simple as "set parameter X" in our SDKs really needs its own section

File: docs-src/python/how-to-asynchronously-complete-an-activity-in-python.md (hunk 13 above)
Comment (flossypurse) [reply docs-src/python/how-to-asynchronously-complete-an-activity-in-python.md:11]: Yes, this is exactly what I called out before as well.

**Review by flossypurse — CHANGES_REQUESTED** [review]

**Review by flossypurse — APPROVED** [review]

File: docs-src/python/how-to-set-a-parent-close-policy-in-python.md [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+tags:
+  - python
+  - developer-guide
+  - how-to
+---
+
+Set the `parent_close_policy` parameter inside the [`start_child_workflow`](https://python.temporal.io/temporalio.workflow.html#start_child_workflow) function or the [`execute_child_workflow()`](https://python.temporal.io/temporalio.workflow.html#execute_child_workflow) function to specify the behavior of the Child Workflow when the Parent Workflow closes.
+
+```python
+async def run(self, name: str) -> str:
```
Comment (cretz) [inline docs-src/python/how-to-set-a-parent-close-policy-in-python.md:15]: FWIW, I don't think we need an entire markdown file, w/ sentences and code snippet and everything, just for showing how to set one parameter.

File: docs-src/typescript/how-to-asynchronously-complete-an-activity-in-typescript.md [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+id: how-to-asynchronously-complete-an-activity-in-typescript
+title: How to asynchronously complete an Activity in TypeScript
+sidebar_label: Asynchronously complete an Activity
+description: To asynchronously complete an Activity in TypeScript, set `AsyncCompletionClient` to `complete`.
+tags:
+  - typescript
+  - how-to
+---
+
+To asynchronously complete an Activity with a Task Token, set the [`AsyncCompletionClient`](https://typescript.temporal.io/api/classes/client.AsyncCompletionClient) class to the [`complete`](https://typescript.temporal.io/api/classes/client.AsyncCompletionClient#complete) method.
```
Comment (lorensr) [inline docs-src/typescript/how-to-asynchronously-complete-an-activity-in-typescript.md:11]:
```suggestion
To asynchronously complete an Activity, call [`AsyncCompletionClient.complete`](https://typescript.temporal.io/api/classes/client.AsyncCompletionClient#complete).
```

File: docs-src/typescript/how-to-set-a-parent-close-policy-in-typescript.md [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+title: How to set Parent Close Policy in Typescript
+sidebar_label: Parent Close Policy
+description: Use the `parentClosePolicy` property.
+tags:
+  - typescript
+  - developer-guide
+  - how-to
+---
+
+To specify how a Child Workflow reacts to a Parent Workflow reaching a Closed state, use the [`parentClosePolicy`](https://typescript.temporal.io/api/interfaces/workflow.ChildWorkflowOptions#parentclosepolicy) property.
```
Comment (lorensr) [inline docs-src/typescript/how-to-set-a-parent-close-policy-in-typescript.md:12]:
```suggestion
To specify how a Child Workflow reacts to a Parent Workflow reaching a Closed state, use the [`parentClosePolicy`](https://typescript.temporal.io/api/interfaces/workflow.ChildWorkflowOptions#parentclosepolicy) option.
```

**Review by lorensr — APPROVED** [review]

**Pull request merged**